source/extensions/clusters/common/logical_host.h

```diff
@@ -61,6 +61,14 @@ class LogicalHost : public HostImpl {
     absl::ReaderMutexLock lock(&address_lock_);
     return HostImpl::address();
   }
+
+  const std::pair<Network::Address::InstanceConstSharedPtr,
+                  const std::vector<Network::Address::InstanceConstSharedPtr>>
+  addressAndListCopy() const {
+    absl::ReaderMutexLock lock(&address_lock_);
```

Comment:
as I've mentioend before I think this pattern of using reader-locks for very short accesses is an anti-pattern.

I will send you an internal article about this separately.